### PR TITLE
Move to an explicit piecestore and explicit unsealing.

### PR DESCRIPTION
**Pull request by hannahhoward**
# Goals

Reduce the node adapter interfaces so they can be satisfied by both go-filecoin and lotus, while also fully supporting CAR files and seperating interface boundaries
 
# Implementation

- Use a piece info state data base to track connections between CommP & deals
- Wait for deal sealing in provider side then store sector info about piece
- The CID Info would eventually be generated when writing the CAR
- Add unsealing explicitly to the retrieval side and add tests

### CODE REVIEW

File: piecestore/piecestore.go
Comment (ingar) [inline piecestore/piecestore.go]: This comment is wondering about whether adding duplicate `DealInfo` (or `BlockInfo` below) is necessarily bad or not.  Since it's a local store maybe it's not a big deal.

File: piecestore/types.go
Comment (ingar) [inline piecestore/types.go]: Are these going to be used in another context?  They are unused here -- or would they be more useful if they didn't return an error?

**Review by ingar — APPROVED** [review]
Some non-blocking comments... 👍

**Comment by hannahhoward** [comment]
@ingar I'm gonna mess with this interface some more in my next PR, cause I realize we actually are gonna have to change some more things... :( so let's not block on it for now.

**Review by shannonwells — APPROVED** [review]
All the suggestions I have are super minor.  Your call

File: retrievalmarket/impl/blockunsealing/blockunsealing.go
Comment (hannahhoward) [reply retrievalmarket/impl/blockunsealing/blockunsealing.go]: ha right

File: retrievalmarket/impl/provider.go
Comment (hannahhoward) [reply retrievalmarket/impl/provider.go]: nah that's copied code that should get deleted :)

Comment (hannahhoward) [reply retrievalmarket/impl/provider.go]: actually i'm not sure will take another look

File: retrievalmarket/impl/blockunsealing/blockunsealing_test.go
Comment (hannahhoward) [reply retrievalmarket/impl/blockunsealing/blockunsealing_test.go]: yea so, this isn't the best test, but the error is actually the same as if it weren't in the blockstore --

if it unsealed a second time, it would work, cause it would put the deleted block back in the block store.

so it's not a specific error -- it just doesn't attempt another unseal -- and then fails reading from the blockstore as a result

File: piecestore/piecestore_test.go
Comment (hannahhoward) [reply piecestore/piecestore_test.go]: yes, this is gonna get changed a bunch shortly so I will hold off for now

**Pull request merged**
